Python-Basics.md

```diff
@@ -0,0 +1,710 @@
+True
+```
+
+## Functions are Objects Too
+
+> In Python, functions are first class objects. 
+
+*Functions* in Python are also *objects*, created with an *identity*, *type*, and *value*. They too can be passed into other *functions*: 
+
+```python
+>>> id(dir)
+4568035688
+>>> type(dir)
+<class 'builtin_function_or_method'>
+>>> dir(dir)
+['__call__', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__module__', '__name__', '__ne__', '__new__', '__qualname__', '__reduce__', '__reduce_ex__', '__repr__', '__self__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__text_signature__']
+>>> issubclass(type(dir), object)
+True
+```
+It is also possible to bind functions to a name and called the bound function using that name:
+
+```python
+>>> a = dir
+>>> print(a)
+<built-in function dir>
+>>> a(a)
+['__call__', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__module__', '__name__', '__ne__', '__new__', '__qualname__', '__reduce__', '__reduce_ex__', '__repr__', '__self__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__text_signature__']
+```
+
+Resources:
+- https://www.jeffknupp.com/blog/2014/06/18/improve-your-python-python-classes-and-object-oriented-programming/
+
+## Variables, Names, and Binding
+
+Having *objects* isn't useful unless there is a way to use them. In order to use an *object*, there must be a way to reference them. In Python this is done by **binding** objects to **names**. A detailed overview of can be found here: https://docs.python.org/3/reference/executionmodel.html
+
+One way this is done is by using an *assignment statement*, https://docs.python.org/3/reference/simple_stmts.html#assignment-statements. This is commonly called *assigning a variable* in the context of Python. If speaking about programming in the context of other languages, **binding** an *object* to a **name** may be more precise.
+
+```python
+>>> some_number = 1
+>>> print(some_number)
+1
+```
+
+In the example above, the target of the assignment statement is a name (identifier), `some_number`. The *object* being assigned is the number 1. The statement **binds** the *object* to the **name**. The second statement, we use this binding `print` the *object* that `some_number` refers to.
+
+The identifier is not preceeded by a *type*. That is because Python is dynamically-typed language. The identifier is bound to an *object* that does have a *type*, however, the identifier itself can be rebound to another *object* of a different *type*:
+
+```python
+>>> some_variable = 1
+>>> print(some_variable)
+1
+>>> some_variable = "Hello campers!"
+>>> print(some_variable)
+Hello campers!
+```
+
+## Keywords are Reserved
+
+Python has a list of keywords that cannot be used as identifiers (variable names):
```

Comment:
Python has a list of [keywords](https://docs.python.org/3/reference/lexical_analysis.html#keywords) that cannot be used as identifiers (variable names).